File: Src/order.api/Domain/Entities/Order.cs
using order.api.Domain.Common;
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace order.api.Domain.Entities
{
    public class Order: AuditableEntity
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        public decimal Total { get; internal set; }

        public decimal TotalItens { get; internal set; }

        [Required]
        public int TableNumber { get; set; }

        [Required]
        public virtual IList<OrderSandwich> Sandwiches { get; set; }

        public virtual IList<OrderPriceRule> Rules { get; set; }

        public OrderStatusEnum Status { get; set; }
    }

}

File: Src/orderApp/src/model/order.ts
import { Ingredient } from './ingredient';
import { Sandwich } from './sandwich';

export class Order {
    id: string;
    tableNumber: number;
    status: number;
    totalItens: number;
    total: number;
    rules: [];
    sandwiches: OrderSandwich[];
  }

  export class NewOrder {
    tableNumber: number;
    totalAdditional: number;
    total: number;
    sandwich: OrderSandwich;
  }

export class OrderSandwich {
    id: string;
    sandwichId: string;
    sandwich: Sandwich;
    position: number;
    quantity: number;
    additionalIngredients: OrderIngredient[];
}

export class OrderIngredient{
    id: string;
    orderSandwichId: string;
    ingredientId: string;
    quantity: number;
    ingredient: Ingredient;
}

